include/gsl/gsl_byte

```diff
@@ -37,10 +37,30 @@
 #pragma push_macro("noexcept")
 #define noexcept /*noexcept*/
 #endif           // _MSC_VER <= 1800
+
+#if _HAS_STD_BYTE 
+
```

Comment:
I suggest:
```c++
#if _MSC_VER >= 1911 && (!defined(_HAS_STD_BYTE) || _HAS_STD_BYTE)
```
Which will avoid trying to use `std::byte` when a user has defined `_HAS_STD_BYTE` to 0 to disable the feature, but will continue to work fine if we rip out the macro entirely.